src/controllers/update/overseas.entity.review.controller.ts

```diff
@@ -27,42 +24,30 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
         appData.entity = {};
       }
       if (!appData.entity.email && overseasEntityId !== undefined && transactionId !== undefined) {
-        const privateOeDetails = await getPrivateOeDetails(req, transactionId, overseasEntityId);
-        if (privateOeDetails === undefined || privateOeDetails.email_address === undefined || privateOeDetails.email_address.length === 0) {
-          const message = `Private OE Details not found for overseas entity ${appData.entity_number}`;
-          logger.errorRequest(req, message);
-          throw new Error(message);
-        }
-        appData.entity.email = privateOeDetails.email_address;
+        try {
+          const privateOeDetails = await getPrivateOeDetails(req, transactionId, overseasEntityId);
+          if (privateOeDetails === undefined || privateOeDetails.email_address === undefined || privateOeDetails.email_address.length === 0) {
+            const message = `Private OE Details not found for overseas entity ${appData.entity_number}`;
+            logger.infoRequest(req, message);
+          } else {
+            appData.entity.email = privateOeDetails.email_address;
 
-        // Cache in session and save out for save&resume.
-        setExtraData(session, appData);
-        await updateOverseasEntity(req, session);
+            // Cache in session and save out for save&resume.
+            setExtraData(session, appData);
+            await updateOverseasEntity(req, session);
+          }
+        } catch (error) {
+          const message = `Private OE Details could not be retrieved for overseas entity ${appData.entity_number}`;
+          logger.infoRequest(req, message);
+        }
       }
     }
 
-    return res.render(config.OVERSEAS_ENTITY_REVIEW_PAGE, {
-      templateName: config.OVERSEAS_ENTITY_REVIEW_PAGE,
-      backLinkUrl,
-      changeLinkUrl,
-      overseasEntityHeading,
-      appData,
-      pageParams: {
-        isRegistration: false
-      },
-    });
-  } catch (errors) {
-    logger.errorRequest(req, errors);
-    next(errors);
-  }
-};
+    return res.redirect(config.OVERSEAS_ENTITY_UPDATE_DETAILS_URL);
```

Comment:
Is there a plan to restore the OVERSEAS_ENTITY_REVIEW_PAGE in future? Otherwise should we not move this logic to the entity controller and get rid of this controller since the html page has been removed

Reply:
No plan as such but the option should be kept open I think. We don't actually have a good solution for this fetch as the previous is an info card and the next is the edit details which is linked from CYA.

Reply:
Happy to add some tech debt to resolve this post MVP.

Reply:
https://companieshouse.atlassian.net/browse/UAR-949